src/main/solution_space/serialized_code.py

```diff
@@ -105,6 +110,12 @@ def tree(self) -> ast.AST:
     def code_info_list(self) -> List[CodeInfo]:
         return self._code_info_list
 
+    def add_next_anon_tree(self, next_anon_tree: 'AnonTree') -> bool:
+        if next_anon_tree.id in self._next_anon_trees_ids:
+            return False
+        self._next_anon_trees_ids.append(next_anon_tree.id)
+        return True
+
```

Comment:
Maybe we should also check that next_anon_tree.id is not self.id? It can happen, but we don't want to pass the same tree in choose_best_anon_tree, so we should be sure that it's not in the next_anon_trees_ids